fast64_internal/f3d/f3d_gbi.py

```diff
@@ -3093,43 +2953,15 @@ def __init__(self, name, x1, y1, x2, y2):
         self.x2 = x2
         self.y2 = y2
 
+    @property
+    def fields(self):
+        return self.x1, self.y1, self.x2, self.y2
+
     def to_binary(self):
-        return (
-            self.x1.to_bytes(4, "big")
-            + self.y1.to_bytes(4, "big")
-            + self.x2.to_bytes(4, "big")
-            + self.y2.to_bytes(4, "big")
-        )
+        return (a.to_bytes(4, "big") for a in self.fields)
```

Comment:
I just realized, the replacement code isn't correct here - it returns a different thing than before. I think this was supposed to be a list comprehension, but even then its returning a list of bytearrays instead of one long bytearray.

Reply:
replaced with
```
    def to_binary(self):
        return b"".join(a.to_bytes(4, "big") for a in self.fields)
```